packages/webpack/src/resource_loader.ts

```diff
@@ -0,0 +1,116 @@
+import {ResourceLoader} from '@angular/compiler';
+import {readFileSync} from 'fs';
+import * as vm from 'vm';
+import * as path from 'path';
+
+const NodeTemplatePlugin = require('webpack/lib/node/NodeTemplatePlugin');
+const NodeTargetPlugin = require('webpack/lib/node/NodeTargetPlugin');
+const LoaderTargetPlugin = require('webpack/lib/LoaderTargetPlugin');
+const SingleEntryPlugin = require('webpack/lib/SingleEntryPlugin');
+
+
+
+export class WebpackResourceLoader implements ResourceLoader {
```

Comment:
Yup it was misunderstanding until I loaded it and tried myself. It makes perfect sense what you are doing.